tests/functional/api/test_merge_requests.py

```diff
@@ -125,10 +125,18 @@ def test_merge_request_should_remove_source_branch(
         time.sleep(0.5)
     assert mr.merged_at is not None
     time.sleep(0.5)
+    result = wait_for_sidekiq(timeout=60)
+    assert result is True, "sidekiq process should have terminated but did not"
 
     # Ensure we can NOT get the MR branch
     with pytest.raises(gitlab.exceptions.GitlabGetError):
-        project.branches.get(source_branch)
+        result = project.branches.get(source_branch)
+        # Help to debug in case the expected exception doesn't happen.
+        import pprint
+
+        print("mr:", pprint.pformat(mr))
+        print("mr.merged_at:", pprint.pformat(mr.merged_at))
+        print("result:", pprint.pformat(result))
```

Comment:
Was this meant to stay here or was it temporary during debugging?

I thought pytest suppressed normal prints so wasn't sure. Maybe we can add `logging` if we want something more long term? Seems like it's natively picked up (`warn` and above): https://docs.pytest.org/en/latest/how-to/logging.html

Reply:
Well I was thinking about leaving it.

1. This code should never be executed if the test works as the call above will raise an exception.
2. If a test fails then pytest will print stdout. If a test succeeds then stdout is suppressed.
3. So if the test fails then it will print the stdout that we generate.

But more than willing to remove it if preferred.

Reply:
Ah, that's right it gets dumped on failure, I forgot. Sure, we can remove it after we finally get rid of the flakiness 😁